### ISSUE TITLE
Prevent Events in specific instances

### ISSUE THREAD

**Issue by retwedt** [500tech/angular2-tree-component#79]
I would like to prevent some events from occurring depending on the context.

An example: a User should not be able to deactivate a node by clicking on it, but the node should still deactivate if the User selects a different node.

Another example:  a User should not be able to toggle the root node, but they should be able to toggle any other nodes (provided they have children of course).

Is there a way I can achieve this?    I am using v1.1.18 (stuck at Angular2 rc5 for the moment).  If so, does it require an upgrade to the newest version of angular2-tree-component?

Thanks for your help/hard work on this fantastic library!

**Comment by adamkleingit** [comment]
Hi!

You can achieve this using `actionMapping.mouse.click` (available in 1.1.18):
https://angular2-tree.readme.io/v1.1.18/docs/action-mapping

The default is TREE_ACTIONS.TOGGLE_SELECTED, which does this:
(tree:TreeModel, node:TreeNode, $event:any) => node.toggleActivated()

You can specify a different method that does something like:
```
(tree:TreeModel, node:TreeNode, $event:any) => {
  if (node.isRoot) return;
  node.setIsActive(true);
}
```

And I strongly recommend you to advance to the release version of Angular 2.
Most open source out there has already advanced, and you'll be stuck behind

All the best!

**Comment by retwedt** [comment]
Thanks for the response!  An Angular2 upgrade is definitely on the list, unfortunately we have a couple of other priorities to get to before then...
